### DISCUSSION TITLE
magic link

### DISCUSSION THREAD

**Discussion by patilrush9** [supabase/supabase#20399] [Questions]
response = supabase_client.auth.create_session(email=email)
response = supabase_client.auth.create_magic_link(email=email)
response = supabase_client.auth.create_sign_in_email(email=email)

all not working i need magic link to send in my custom email so user can login from that email and we can proceed to further processes
we need validate user and loged in user to do some processes

**Comment by GaryAustin1** [comment]
You may need to provide more info on what you are trying to do.
All auth calls are here: https://supabase.com/docs/reference/javascript/auth-api
The only two calls I can think of that would be relevant to magic link are:
signInWithOtp https://supabase.com/docs/reference/javascript/auth-signinwithotp
admin.generateLink https://supabase.com/docs/reference/javascript/auth-admin-generatelink

You can use generateLink to get URL's and tokens for your own email.  This is done serverside with service_role key.

**Reply by patilrush9** [reply]
my basic need is i want to login in to my any user and check there issue so i need to do you have any option for that like master password ,magic link or create Bearer token genrating from Client = create_client(url, key) this client in python
